pkg/resources/connection.go

```diff
@@ -56,6 +56,13 @@ func ResourceConnection() *schema.Resource {
 				Required:     true,
 				Description:  "The type of connection",
 				ValidateFunc: validation.StringInSlice(connectionTypes, false),
+				ForceNew:     true,
```

Comment:
Just double checking this is intentionally on type rather than private link? (Just as that's what this PR relates to)